pages/api/v1/verification/twitter-callback.ts

```diff
@@ -60,42 +62,72 @@ async function getOAuthAccessTokenWith({
 	});
 }
 
-const handler: NextApiHandler<any> = async (req, res) => {
-	const {
-		oauth_verifier: oauthVerifier,
-		oauthRequestToken,
-		oauthRequestTokenSecret
-	} = req.query;
+export const getTwitterCallback = async({ network, oauthVerifier, oauthRequestToken }: Props ): Promise<IApiResponse<string | MessageType>> => {
+	try{
+		const tokenVerification = await firestore.collection('twitter_verification_tokens').where('oauth_request_token', '==', oauthRequestToken).limit(1).get();
+		if (tokenVerification.empty) {
+			return {
+				data: null,
+				error: 'Wrong verification token found!',
+				status: 400
+			};
+		}
+		const data = tokenVerification.docs[0].data();
 
-	const network = String(req.headers['x-network']);
-	if (!network || !isValidNetwork(network)) return res.status(400).json({ message: 'Invalid network in request header' });
+		const oauthRequestTokenSecret = data?.oauth_requestToken_secret;
 
-	const { oauthAccessToken, oauthAccessTokenSecret, results } =
-	await getOAuthAccessTokenWith({
-		oauthRequestToken,
-		oauthRequestTokenSecret,
-		oauthVerifier
-	});
+		const { oauthAccessToken, oauthAccessTokenSecret, results } =
+		await getOAuthAccessTokenWith(network,{ oauthRequestToken, oauthRequestTokenSecret, oauthVerifier });
 
-	const { user_id: userId /*, screen_name */ } = results;
-	const user = await oauthGetUserById(userId, {
-		oauthAccessToken,
-		oauthAccessTokenSecret
-	});
+		const { user_id: userId /*, screen_name */ } = results;
+		const user = await oauthGetUserById(userId, network,
+			{
+				oauthAccessToken,
+				oauthAccessTokenSecret
+			});
 
-	const twitterVerification = await firestore.collection('twitter_verification_tokens').doc(user.screen_name || '').get();
-	const data = twitterVerification.data();
+		const twitterVerificationRef = firestore.collection('twitter_verification_tokens').doc(user?.screen_name);
```

Comment:
- get doc with same requestToken
- check if screen name stored in doc  === user.screen_name
- if yes set verified = true
- else error : invaid username provided or somthing to the same effect